xchange-coindirect/src/main/java/info/bitrich/xchangestream/coindirect/service/CoindirectStreamingMarketDataService.java

```diff
@@ -0,0 +1,144 @@
+
+        return service.subscribeChannel(channelName)
+                .map(s -> {
+                    CoindirectOrderbook coindirectOrderbook = coindirectOrderbookMap.get(currencyPair);
+                    OrderBook orderBook = orderBookMap.get(currencyPair);
+
+                    if (coindirectOrderbook == null) {
+
+                        coindirectOrderbook = getCoindirectMarketDataServiceRaw().getCoindirectOrderbook(currencyPair);
+                        coindirectOrderbookMap.put(currencyPair, coindirectOrderbook);
+
+                        orderBook = CoindirectAdapters.adaptOrderBook(currencyPair, coindirectOrderbook);
+                        orderBookMap.put(currencyPair, orderBook);
+                    }
+
+                    CoindirectOrderBookEvent coindirectOrderBookEvent = objectMapper.readValue(s, CoindirectOrderBookEvent.class);
+
+                    if (coindirectOrderBookEvent.sequence > coindirectOrderbook.sequence) {
+                        OrderBookUpdate orderBookUpdate = new OrderBookUpdate(
+                                CoindirectAdapters.convert(coindirectOrderBookEvent.side),
+                                coindirectOrderBookEvent.size,
+                                currencyPair,
+                                coindirectOrderBookEvent.price,
+                                new Date(),
+                                coindirectOrderBookEvent.size
+                        );
+                        orderBook.update(orderBookUpdate);
+                    } else {
+                            /* Ignore this event as its older than the latest sequence */
+                    }
+
+                    return orderBook;
+                });
+    }
+
+    @Override
+    public Observable<Ticker> getTicker(CurrencyPair currencyPair, Object... args) {
+        String channelName = "ticker-" + CoindirectAdapters.toSymbol(currencyPair);
+        return service.subscribeChannel(channelName).map(s -> {
+            CoindirectTradeEvent coindirectTradeEvent = objectMapper.readValue(s, CoindirectTradeEvent.class);
+            return new Ticker.Builder()
+                .currencyPair(currencyPair)
+                .last(coindirectTradeEvent.price)
+                .volume(coindirectTradeEvent.amount)
+                .timestamp(new Date(coindirectTradeEvent.timestamp))
+                .build();
+        });
+    }
+
+    @Override
+    public Observable<Trade> getTrades(CurrencyPair currencyPair, Object... args) {
+        String channelName = "ticker-" + CoindirectAdapters.toSymbol(currencyPair);
+        return service.subscribeChannel(channelName).map(s -> {
+            CoindirectTradeEvent coindirectTradeEvent = objectMapper.readValue(s, CoindirectTradeEvent.class);
+            return new Trade(Order.OrderType.BID, coindirectTradeEvent.amount, currencyPair, coindirectTradeEvent.price, new Date(coindirectTradeEvent.timestamp), null);
+        });
+    }
+
+    public Observable<Trade> getAccountTrades() {
```

Comment:
Rename to `getTrades` for consistency

Reply:
Thanks

Reply:
This belongs on a `StreamingTradeService` implementation